src/order/bounded_order.lean

```diff
@@ -1130,110 +1132,96 @@ instance [has_le α] [has_le β] [bounded_order α] [bounded_order β] : bounded
+@[simp] lemma min_eq_top [order_top α] {a b : α} : min a b = ⊤ ↔ a = ⊤ ∧ b = ⊤ := inf_eq_top_iff
+
+end linear_order
+
 /-! ### Disjointness and complements -/
 
 section disjoint
 section semilattice_inf_bot
-variables [semilattice_inf α] [order_bot α]
+variables [semilattice_inf α] [order_bot α] {a b c d : α}
 
 /-- Two elements of a lattice are disjoint if their inf is the bottom element.
   (This generalizes disjoint sets, viewed as members of the subset lattice.) -/
 def disjoint (a b : α) : Prop := a ⊓ b ≤ ⊥
 
-theorem disjoint.eq_bot {a b : α} (h : disjoint a b) : a ⊓ b = ⊥ :=
-eq_bot_iff.2 h
-
-theorem disjoint_iff {a b : α} : disjoint a b ↔ a ⊓ b = ⊥ :=
-eq_bot_iff.symm
-
-theorem disjoint.comm {a b : α} : disjoint a b ↔ disjoint b a :=
-by rw [disjoint, disjoint, inf_comm]
+lemma disjoint_iff : disjoint a b ↔ a ⊓ b = ⊥ := le_bot_iff
+lemma disjoint.eq_bot : disjoint a b → a ⊓ b = ⊥ := bot_unique
+lemma disjoint.comm : disjoint a b ↔ disjoint b a := by rw [disjoint, disjoint, inf_comm]
+@[symm] lemma disjoint.symm ⦃a b : α⦄ : disjoint a b → disjoint b a := disjoint.comm.1
+lemma symmetric_disjoint : symmetric (disjoint : α → α → Prop) := disjoint.symm
+lemma disjoint_assoc : disjoint (a ⊓ b) c ↔ disjoint a (b ⊓ c) :=
+by rw [disjoint, disjoint, inf_assoc]
 
-@[symm] theorem disjoint.symm ⦃a b : α⦄ : disjoint a b → disjoint b a :=
-disjoint.comm.1
+@[simp] lemma disjoint_bot_left : disjoint ⊥ a := inf_le_left
+@[simp] lemma disjoint_bot_right : disjoint a ⊥ := inf_le_right
 
-lemma symmetric_disjoint : symmetric (disjoint : α → α → Prop) := disjoint.symm
+lemma disjoint.mono (h₁ : a ≤ b) (h₂ : c ≤ d) : disjoint b d → disjoint a c :=
+le_trans $ inf_le_inf h₁ h₂
 
-@[simp] theorem disjoint_bot_left {a : α} : disjoint ⊥ a := inf_le_left
-@[simp] theorem disjoint_bot_right {a : α} : disjoint a ⊥ := inf_le_right
+lemma disjoint.mono_left (h : a ≤ b) : disjoint b c → disjoint a c := disjoint.mono h le_rfl
+lemma disjoint.mono_right : b ≤ c → disjoint a c → disjoint a b := disjoint.mono le_rfl
 
-theorem disjoint.mono {a b c d : α} (h₁ : a ≤ b) (h₂ : c ≤ d) :
-  disjoint b d → disjoint a c := le_trans (inf_le_inf h₁ h₂)
+variables (c)
 
-theorem disjoint.mono_left {a b c : α} (h : a ≤ b) : disjoint b c → disjoint a c :=
-disjoint.mono h le_rfl
+lemma disjoint.inf_left (h : disjoint a b) : disjoint (a ⊓ c) b := h.mono_left inf_le_left
+lemma disjoint.inf_left' (h : disjoint a b) : disjoint (c ⊓ a) b := h.mono_left inf_le_right
+lemma disjoint.inf_right (h : disjoint a b) : disjoint a (b ⊓ c) := h.mono_right inf_le_left
+lemma disjoint.inf_right' (h : disjoint a b) : disjoint a (c ⊓ b) := h.mono_right inf_le_right
 
-theorem disjoint.mono_right {a b c : α} (h : b ≤ c) : disjoint a c → disjoint a b :=
-disjoint.mono le_rfl h
+variables {c}
 
-@[simp] lemma disjoint_self {a : α} : disjoint a a ↔ a = ⊥ :=
-by simp [disjoint]
+@[simp] lemma disjoint_self : disjoint a a ↔ a = ⊥ := by simp [disjoint]
```

Comment:
I want to call it `disjoint.eq_bot`, but the name is already taken. What about renaming the current `disjoint.eq_bot` to `disjoint.inf_eq`? This will probably be a separate PR.

Reply:
Sure, but please add an issue for this or a TODO line for it, so it doesn't get forgotten about.